codex/blockexchange/engine/engine.nim

```diff
@@ -167,38 +173,38 @@ proc monitorBlockHandle(b: BlockExcEngine, handle: Future[Block], address: Block
 proc requestBlock*(
   b: BlockExcEngine,
   address: BlockAddress,
-  timeout = DefaultBlockTimeout): Future[Block] {.async.} =
+  timeout = DefaultBlockTimeout): Future[?!Block] {.async.} =
```

Comment:
I would still very much like to know why this became a result[block]. It seems to make no difference here nor in the place where this is called. What's going on, and how does it relate to wiring in the prover?

Reply:
Handling results, rather than exceptions is a lot simpler and it doesn't break cancellations with `.catch` at the callsite. This should have been a result all along.

Reply:
I agree, it should have been a result all along. If this change is not related to the prover, then this is some clean-up. It doesn't necessarily belong in this PR. But I don't insist it should be taken out. I defer to your judgement on this! :D. I am in favor of doing small clean-ups as we go. I wanted to be sure I wasn't missing something about the prover integration.